recipes/_install.rb

```diff
@@ -55,23 +55,39 @@ def nsp
   only_if(&manifest_missing)
 end
 
-package node['splunk']['package']['base_name'] do
-  source splunk_file
-  version "#{node['splunk']['package']['version']}-#{node['splunk']['package']['build']}"
-  provider node['splunk']['package']['provider']
-  only_if(&manifest_missing)
-  if platform_family?('windows')
+if platform_family? 'rhel', 'fedora', 'amazon'
+  rpm_package node['splunk']['package']['base_name'] do
+    source splunk_file
+    version "#{node['splunk']['package']['version']}-#{node['splunk']['package']['build']}"
+    only_if(&manifest_missing)
+  end
+elsif platform_family? 'debian'
+  dpkg_package node['splunk']['package']['base_name'] do
+    source splunk_file
+    version "#{node['splunk']['package']['version']}-#{node['splunk']['package']['build']}"
+    only_if(&manifest_missing)
+  end
+elsif platform_family? 'windows'
+  windows_package node['splunk']['package']['base_name'] do
+    source splunk_file
+    version "#{node['splunk']['package']['version']}-#{node['splunk']['package']['build']}"
+    only_if(&manifest_missing)
     # installing as the system user by default as Splunk has difficulties with being a limited user
     options %(AGREETOLICENSE=Yes SERVICESTARTTYPE=auto LAUNCHSPLUNK=0 INSTALLDIR="#{node['splunk']['home'].tr('/', '\\')}")
   end
+else
+  fail 'unsupported platform'
 end
 
 include_recipe 'cerner_splunk::_configure_secret'
 
+windows_password = CernerSplunk::DataBag.load(node['splunk']['windows_password'], secret: node['splunk']['data_bag_secret'])
```

Comment:
`node['splunk']['windows_password']` needs to be added to the attributes doc.

Reply:
added in ebac2d9